airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java

```diff
@@ -180,41 +182,45 @@ public static Map<AirbyteStreamNameNamespacePair, CursorBasedStatus> getCursorBa
+    final List<io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair> failedToQuery = new ArrayList<>();
     streams.forEach(stream -> {
-      final AirbyteStreamNameNamespacePair namespacePair =
-          new AirbyteStreamNameNamespacePair(stream.getStream().getName(), stream.getStream().getNamespace());
-      final long l = fileNodeForStreams(database, namespacePair, quoteString);
-      fileNodes.put(namespacePair, l);
+      try {
+        final AirbyteStreamNameNamespacePair namespacePair =
+            new AirbyteStreamNameNamespacePair(stream.getStream().getName(), stream.getStream().getNamespace());
+        final long l = fileNodeForStreams(database, namespacePair, quoteString);
+        fileNodes.put(namespacePair, l);
+      } catch (final Exception e) {
+        LOGGER.warn("Failed to fetch relation node for {}.{} .", stream.getStream().getNamespace(), stream.getStream().getName(), e);
+        failedToQuery.add(io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair.fromConfiguredAirbyteSteam(stream));
+      }
     });
-    return fileNodes;
+    return new ResultWithFailed<>(fileNodes, failedToQuery);
   }
 
-  public static long fileNodeForStreams(final JdbcDatabase database, final AirbyteStreamNameNamespacePair stream, final String quoteString) {
-    try {
-      final String streamName = stream.getName();
-      final String schemaName = stream.getNamespace();
-      final String fullTableName =
-          getFullyQualifiedTableNameWithQuoting(schemaName, streamName, quoteString);
-      final List<JsonNode> jsonNodes = database.bufferedResultSetQuery(
-          conn -> conn.prepareStatement(CTID_FULL_VACUUM_REL_FILENODE_QUERY.formatted(fullTableName)).executeQuery(),
-          resultSet -> JdbcUtils.getDefaultSourceOperations().rowToJson(resultSet));
-      Preconditions.checkState(jsonNodes.size() == 1);
-      final long relationFilenode = jsonNodes.get(0).get("pg_relation_filenode").asLong();
-      LOGGER.info("Relation filenode is for stream {} is {}", fullTableName, relationFilenode);
-      return relationFilenode;
-    } catch (final SQLException e) {
-      throw new RuntimeException(e);
-    }
+  public static long fileNodeForStreams(final JdbcDatabase database, final AirbyteStreamNameNamespacePair stream, final String quoteString)
+      throws SQLException {
+    final String streamName = stream.getName();
+    final String schemaName = stream.getNamespace();
+    final String fullTableName =
+        getFullyQualifiedTableNameWithQuoting(schemaName, streamName, quoteString);
+    final List<JsonNode> jsonNodes = database.bufferedResultSetQuery(
+        conn -> conn.prepareStatement(CTID_FULL_VACUUM_REL_FILENODE_QUERY.formatted(fullTableName)).executeQuery(),
+        resultSet -> JdbcUtils.getDefaultSourceOperations().rowToJson(resultSet));
+    Preconditions.checkState(jsonNodes.size() == 1);
+    final long relationFilenode = jsonNodes.get(0).get("pg_relation_filenode").asLong();
```

Comment:
nit: for a view file node is NULL, so technically you can check for NULL and change the function signature to Optional<Long> to avoid a null exception thrown at runtime. Marking as nit because callee is doing a try-catch on this function. 

Ref: from https://www.postgresql.org/docs/9.0/functions-admin.html  about pg_relation_filenode  `The function returns NULL if passed a relation that does not have storage, such as a view.` from

Reply:
it is null.
Thanks. Got rid of the error